### PR TITLE
doc: add the migration guide to the new guide series

### PR DESCRIPTION
**Pull request by mspncp**
Renames `migration_guide(7)` and adds it as new entry
```
=item L<ossl-guide-migration(7)>: Migrating from older OpenSSL versions
```
to `ossl-guide-introduction(7)` .

### CODE REVIEW

File: doc/man7/ossl-guide-migration.pod [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 =head1 NAME
 
-migration_guide - OpenSSL migration guide
+ossl-guide-migration, migration_guide
```
Comment (mspncp) [inline doc/man7/ossl-guide-migration.pod:5]: Keeps `migration_guide` as alias for compatibility reasons.

**Comment by mattcaswell** [comment]
It's just different to how "make update" wants it formatted. So you just run "make update" again and check in the changes.

**Comment by mspncp** [comment]
Thanks for the hint. I recalled that I need to run `make update` shortly after posting my question.

**Review by hlandau — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by hlandau** [comment]
Merged to master. Thank you.

**Pull request closed without merging**